addons/mail/static/src/views/web/list/archive_disabled_list_controller.js

```diff
@@ -0,0 +1,10 @@
+/** @odoo-module */
+
+import { ListController } from "@web/views/list/list_controller";
+
+export class ArchiveDisabledListController extends ListController {
+    setup() {
+        super.setup();
+        this.archiveEnabled = false;
```

Comment:
I see that in list controller
```
        this.archiveEnabled =
            "active" in this.props.fields
                ? !this.props.fields.active.readonly
            ...
```

If we set the field as readonly in the view, isn't it working automatically ?

Reply:
That was what i was hoping but unfortunately it wasn’t so I have create that controller (I mention it in the commit)

Reply:
I'm curious now, why is this not working? Would make sense to disable the button if 'active' is set as readonly indeed.

Reply:
Maybe I have missed something but it seems there is no option to disable archive action:
```javascript
export function getActiveActions(rootNode) {
    return {
        type: "view",
        edit: archParseBoolean(rootNode.getAttribute("edit"), true),
        create: archParseBoolean(rootNode.getAttribute("create"), true),
        delete: archParseBoolean(rootNode.getAttribute("delete"), true),
        duplicate: archParseBoolean(rootNode.getAttribute("duplicate"), true),
    };
}
```
In the list controller, we have:
```javascript
this.archiveEnabled =
    "active" in this.props.fields
        ? !this.props.fields.active.readonly
        : "x_active" in this.props.fields
        ? !this.props.fields.x_active.readonly
        : false;
```
this.props.fields seems to be the field definition of the model because all field are present, not only the one present in the view.
So adding a readonly field in the view doesn't disable the archive/unarchive actions:
```xml
<record id="mail_activity_view_tree_without_record_access" model="ir.ui.view">
    <field name="name">mail.activity.view.tree.without.record.access</field>
    <field name="model">mail.activity</field>
    <field name="inherit_id" ref="mail_activity_view_tree"/>
    <field name="mode">primary</field>
    <field name="priority">32</field>
    <field name="arch" type="xml">
        <xpath expr="//tree" position="inside">
            <field name="active" invisible="1" readonly="1"/>
        </xpath>
    </field>
</record>
```
I think we could improve that (modifying list_arch_parser.js to include that option, or taking into account the readonly option on the active field) and then remove that widget. What do you think ?

Reply:
Ah I see, it only considers the actual field definition and not the XML attributes.

Well that is sad, clearly something that could be improved on the web client side I feel like.

It's ok though, clearly not a big deal, let us keep it as is then.

Thank you for the detailed explanations!